src/ja/server/dispatcher/Dispatcher.py

```diff
@@ -0,0 +1,36 @@
+from ja.common.job import Job
+
+
+class Dispatcher:
+    """
+    Represents the mediator between the Scheduler and the work machines.
+    """
+
+    def __init___(self):
+        """!
+        Constructor for the dispatcher class
+        """
+
+    def set_distribution(self, jobs: list[Job]) -> None:
+        """!
+        Distributes the newly added jobs to the work machines
+        and potentially pauses jobs that need to be paused.
+
+        @param jobs list of the new job distribution.
+        """
+
+    def _get_changed_jobs(self) -> list[Job]:
+        """!
+        @return list of the jobs whose state has been changed.
+        """
+
+    def _reset(self) -> None:
+        """!
+        Dispatches the added jobs, pauses the paused jobs and cancels the cancelled jobs.
+        """
+
+    @property
+    def jobs(self):
```

Comment:
Why is this property needed? I think that the scheduler simply gives the dispatcher the next distribution, there's no need to get the information from the dispatcher.